### PR TITLE
issue: 848172 fix csbuild for 'std::bitset'

### PR DESCRIPTION
**Pull request by rosenbaumalex**
convert 'memset(struct)' to 'struct::reset()'

(CPPCHECK_WARNING) error[memsetClass]: Using 'memset' on struct that contains a 'std::bitset'

Signed-off-by: Alex Rosenbaum <Alexr@mellanox.com>

### CODE REVIEW

**Comment by rosenbaumalex** [comment]
@liranoz12 please review

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/405/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/419/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/406/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/420/ for details (Mellanox internal link).

File: src/vma/util/vma_stats.h [hunk 1]
```diff
@@ -303,19 +324,38 @@ typedef struct sh_mem_t {
...
 	cq_instance_block_t		cq_inst_arr[NUM_OF_SUPPORTED_CQS];
 	ring_instance_block_t		ring_inst_arr[NUM_OF_SUPPORTED_RINGS];
 	bpool_instance_block_t		bpool_inst_arr[NUM_OF_SUPPORTED_BPOOLS];
 	mc_grp_info_t			mc_info;
 	iomux_stats_t			iomux;
-	int				fd_dump;
-	vlog_levels_t			fd_dump_log_level;
 
 	// MUST BE LAST ENTRY in struct: [0] is the allocation start point for all fd's
+	size_t				max_skt_inst_num;
```
Comment (liranoz12) [inline src/vma/util/vma_stats.h]: I think we should replace line 338 with line 337

File: src/vma/util/vma_stats.h [hunk 2]
```diff
@@ -232,11 +234,30 @@ typedef struct {
...
+		socket_type = 0;
+		b_is_offloaded = b_blocking = b_mc_loop = false;
+		bound_if = connected_ip = mc_tx_if = (in_addr_t)0;
+		bound_port = connected_port = (in_port_t)0;
+		threadid_last_rx = threadid_last_tx = pid_t(0);
+		n_rx_ready_pkt_count = n_rx_ready_byte_count = n_rx_ready_byte_limit = n_rx_zcopy_pkt_count = n_tx_ready_byte_count = 0;
+		memset(&counters, 0, sizeof(counters));
+		mc_grp_map.reset();
+	};
 } socket_stats_t;
```
Comment (liranoz12) [inline src/vma/util/vma_stats.h]: It is better to memset all the struct except the bistset elemet :
```
memset(this, 0, sizeof(socket_stats_t) - sizeof(mc_grp_map));
mc_grp_map.reset();
```
We just need to add comment that the bitset must be the last elemet in the struct

File: src/vma/util/vma_stats.h [hunk 3]
```diff
@@ -358,9 +398,4 @@ void print_full_stats(socket_stats_t* p_si_stats, mc_grp_info_t* p_mc_grp_info,
 void print_netstat_like(socket_stats_t* p_si_stats, mc_grp_info_t* p_mc_grp_info, FILE* file, int pid);
 void print_netstat_like_headers(FILE* file);
```
Comment (liranoz12) [inline src/vma/util/vma_stats.h]: You can also remove the` <bitset>` includes from :
stats_reader.cpp : line 44
stats_printer.cpp : line 44

File: src/vma/util/vma_stats.h (hunk 1 above)
Comment (rosenbaumalex) [inline src/vma/util/vma_stats.h]: will update and add comment

File: src/vma/util/vma_stats.h (hunk 3 above)
Comment (rosenbaumalex) [inline src/vma/util/vma_stats.h]: will remove extra inlucdes

File: src/vma/util/vma_stats.h (hunk 2 above)
Comment (rosenbaumalex) [inline src/vma/util/vma_stats.h]: I will not do this change now. maybe as a later PR

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/409/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/423/ for details (Mellanox internal link).

**Pull request merged**